spec/requests/application_controller_spec.rb

```diff
@@ -143,6 +143,22 @@
         expect(response.status).to eq(404)
         expect(response.body).to_not include('google.com/search')
       end
+
+      it 'should cache results' do
+        $redis.flushall
```

Comment:
Small one but we usually assume that each test cleans up after itself. Therefore, there shouldn't be a need to flush redis here.

Reply:
> A quick look for expect(response.status).to eq(404) returns 122 matches.

Hmm that does mean that all those tests are already leaking a state in Redis. I can't think of a clean way to handle this right now without using some sort of mock Redis like you mentioned xD. Perhaps we can set the `TTL` for the key to be really low in the test env? We leak state but we leak it for a much shorter period of time.

Reply:
Yeah lets not flushall in the test, instead expose a helper method from the controller that the test can consume to clean up the cache first. It is a good API to have anyway.

Reply:
`$redis.del("page_not_found_topics")` seems to be better. I do not see any point in creating a different method that would only be used by the test.

Reply:
Was there a change made for this? It seems like the we’re not cleaning up the cache properly for the other tests that results in a 404.